dracut/modules.d/99kiwi-dump-reboot/kiwi-dump-reboot-system.sh

```diff
@@ -33,17 +33,36 @@ function boot_installed_system {
     fi
 }
 
+function mount_sysroot {
+    local image_target
+    local root_uuid
+
+
+    [ -e /config.bootoptions ] || die "Missing config.bootoptions file"
+
+    root_uuid=$(
+        while read -r -d ' ' opt; do echo "${opt}";done < /config.bootoptions |\
+        grep root= | cut -f2- -d=
+    )
+
+    [ -z "${root_uuid}" ] && \
+        die "'root=' argument not found in config.bootoptions"
+
+
+    image_target=$(get_selected_disk)
+    kpartx -s -a "${image_target}"
+    systemd-mount --fsck=no --no-block "${root_uuid}" /sysroot
+}
+
 #======================================
 # Reboot into system
 #--------------------------------------
 
 if getargbool 0 rd.kiwi.ramdisk; then
     # For ramdisk deployment a kexec boot is not possible as it
     # will wipe the contents of the ramdisk. Therefore we prepare
-    # the switch_root from this deployment initrd. Also see the
-    # unit generator: dracut-kiwi-ramdisk-generator
-    image_target=$(get_selected_disk)
-    kpartx -s -a "${image_target}"
+    # the switch_root from this deployment initrd.
+    mount_sysroot
```

Comment:
So mount_sysroot is called in the ramdisk case only. The code from here has been moved into mount_sysroot and hides the scope. It feels like mount_sysroot can be used generically, but this is not the case. The mount_sysroot method itself contains code parts from here and code parts from PXE. Sorry I'm very confused about all that

Reply:
Right I agree the method name is missleading. This method is ramdisk deployment specific, the method name should reflect this fact. What about `mount_ramdisk`?

Reply:
Also not that my patch on affects the ramdisk deployment execution path

Reply:
yes please rename it, mount_ramdisk_sysroot or similar would be good imho